spk/gitea/src/service-setup.sh

```diff
@@ -5,39 +5,42 @@ PATH="/var/packages/git/target/bin:${PATH}"
 if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 7 ]; then
     SYNOPKG_PKGHOME="${SYNOPKG_PKGDEST}"
 fi
+
+if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
+    SYNOPKG_PKGHOME="${SYNOPKG_PKGVAR}"
+fi
+
 ENV="PATH=${PATH} HOME=${SYNOPKG_PKGHOME}"
 
 SERVICE_COMMAND="env ${ENV} ${GITEA} web --port ${SERVICE_PORT} --pid ${PID_FILE}"
 SVC_BACKGROUND=y
 
-service_preinst ()
+service_postinst ()
 {
-    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ] && [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
-        SHARED_FOLDER="${wizard_shared_folder_name}"
-        if [ ! -d "${SHARED_FOLDER}" ]; then
-            mkdir -p "${SHARED_FOLDER}" || {
-                echo "Failed to create directory \"${SHARED_FOLDER}\"."
-                exit 1
-            }
-        fi
-        set_syno_permissions "${SHARED_FOLDER}" "${EFF_USER}"
+    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
+        IP=$(ip route get 1 | awk '{print $(NF);exit}')
     fi
 }
 
-service_postinst ()
+service_preupgrade ()
 {
-    if [ "${SYNOPKG_PKG_STATUS}" == "INSTALL" ]; then
-        SHARED_FOLDER="${wizard_shared_folder_name}"
-        IP=$(ip route get 1 | awk '{print $(NF);exit}')
-        # Default configuration with shared folder
-        {
-            echo "[repository]"
-            echo "ROOT = ${SHARED_FOLDER}/gitea-repositories"
-            echo "[server]"
-            echo "LFS_CONTENT_PATH = ${SHARED_FOLDER}/lfs"
-            echo "SSH_DOMAIN = ${IP:=localhost}"
-            echo "DOMAIN = ${IP:=localhost}"
-            echo "ROOT_URL = http://${IP:=localhost}:${SERVICE_PORT}/"
-        } > "$CONF_FILE"
+    #Backup existing config on DSM6
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 6 ]; then
+        if [-f "${SYNOPKG_PKGHOME}/conf.ini" ]; then
+            echo "Backup old config on DSM6"  
+            mv ${SYNOPKG_PKGHOME}/conf.ini ${SYNOPKG_PKGHOME}/conf.ini.bck 2>&1
+        fi
     fi
 }
+
+service_postupgrade ()
+{
+    #Restore existing config on DSM6
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 6 ]; then
+        if [-f "${SYNOPKG_PKGHOME}/conf.ini.bkc" ]; then
+            echo "Restore old config on DSM6" 
+            rm -f ${SYNOPKG_PKGHOME}/conf.ini
+            mv ${SYNOPKG_PKGHOME}/conf.ini.bck ${SYNOPKG_PKGHOME}/conf.ini 2>&1
+        fi
+    fi
+}
```

Comment:
please use `SYNOPKG_PKGVAR` here and not `SYNOPKG_PKGHOME`.

you have to define `HOME=${SYNOPKG_PKGHOME}` and since `SYNOPKG_PKGHOME` is not defined on DSM < 7, you define `SYNOPKG_PKGHOME` as `SYNOPKG_PKGVAR` for DSM < 7.

but the config file is in `${SYNOPKG_PKGVAR}` for all DSM versions (otherwise it is very confusing...).

---

And you don't need this save and restore code.
have a look at `service_restore` in `spk/redis/src/service-setup.sh`

Sorry for my misleading (wrong) information. The `conf.ini` file will not be overwritten on upgrades for DSM < 7.  Only the probably updated conf.ini file in the upgrade package ist lost, so the script has to rename it (*.new) before it is overwritten by the restore from TMP folder (code of synocommunity generic install code).
This is good practice, so users may see what has changed by `diff conf.ini conf.ini.new` after package upgrade.

Solution:
remove `service_preupgrade` and `service_postupgrade` and add `service_restore` like this
``` bash
# service_restore is called by post_upgrade before restoring files from ${TMP_DIR}
service_restore ()
{
    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
        # make a copy of the new config file before it gets overwritten by restore
        # overwrite existing *.new files in ${TMP_DIR}/ as all files in ${TMP_DIR}/
        # are restored to ${SYNOPKG_PKGVAR}/
        [ -f "${SYNOPKG_PKGVAR}/conf.ini" ] && cp -f ${SYNOPKG_PKGVAR}/conf.ini ${TMP_DIR}/conf.ini.new
    fi
}
```

As you can see in the comment, the new config.ini file is not renamed in the var folder but copied with the *.new name to `${TMP_DIR}/` (the backup folder) to not get overwritten by restore of outdated conf.ini.new file from former update...

Reply:
Thank you for review, removed additional preupgrade and postupgrade  steps and added service_restore like in redis package